eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/utils/spanned/EmptyLinesTrimming.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.utils.spanned;
+
+import android.text.SpannableStringBuilder;
+import android.text.Spanned;
+
+import com.schedjoules.eventdiscovery.framework.utils.factory.Factory;
+
+
+/**
+ * {@link Spanned} decorator that trims leading and trailing empty lines.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class EmptyLinesTrimming extends AbstractSpanned
```

Comment:
This was taken from the snippet you posted and which I saw myself, too, somewhere, so just to have a well enough solution for this. I am a bit unsure about some parts of this spanned, charsequence trimming, so I didn't want to spend more time with this task to look into them more thoroughly.
Things like: which are the whitespace char codes, how best to check for them,
how does spanned work internally, is it possibly that it uses some non-visible chars for formatting purposes, can that cause problem if we strip it,
why is that that subsequence() didn't seem to work with spanned (lost styling) (this code snippet uses delete() ),
if we want to trim new lines, do we check for '\n' or '\r' as well.
Writing unit test for anything new also adds to the time.

So now that server-side might implement trimming https://github.com/schedjoules/event-discovery-backend/issues/45 , my question is, should we spend more time on this now or not?
We could use this version, and put a task to backlog to revisit it after server-side trimming has been implemented. At that time, I think we only need to trim trailing new lines for html spanned, nothing else, to remove those lines added by `Html` for paragraph endings.